google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/EnhancedBigtableStub.java

```diff
@@ -236,6 +236,38 @@ public static ClientContext createClientContext(EnhancedBigtableStubSettings set
     return ClientContext.create(builder.build());
   }
 
+  private static ImmutableList<ClientInterceptor> getInterceptors(
+      EnhancedBigtableStubSettings.Builder settings) {
+    Set<ConnectionErrorCountInterceptor> connectionErrorCountInterceptors =
+        setupConnectionErrorCountInterceptors(settings);
+    ConnectionErrorCountInterceptor connectionErrorCountInterceptor =
+        new ConnectionErrorCountInterceptor();
+    connectionErrorCountInterceptors.add(connectionErrorCountInterceptor);
+    ImmutableList.Builder<ClientInterceptor> builder =
+        ImmutableList.<ClientInterceptor>builder().add(connectionErrorCountInterceptor);
+
+    if (settings.getEnableRoutingCookie()) {
+      // TODO: this also need to be added to BigtableClientFactory
+      builder.add(new CookiesInterceptor());
+    }
+
+    return builder.build();
+  }
+
+  private static Set<ConnectionErrorCountInterceptor> setupConnectionErrorCountInterceptors(
```

Comment:
I think the naming of this method is a bit confusing, this is really set up the background task instead of setting up the interceptors. Let's rename this method to "setupConnectionErrorCountTask"